cmd/clusterctl/pkg/client/cluster/move_test.go

```diff
@@ -0,0 +1,636 @@
+							"apiVersion": "a/v1",
+							"kind":       "A",
+							"metadata": map[string]interface{}{
+								"namespace": "ns",
+								"name":      "foo",
+								"uid":       "1",
+								"ownerReferences": []interface{}{
+									map[string]interface{}{
+										"apiVersion": "b/v1",
+										"kind":       "B",
+										"name":       "bar",
+										"uid":        "2",
+									},
+								},
+							},
+						},
+					},
+					{
+						Object: map[string]interface{}{
+							"apiVersion": "b/v1",
+							"kind":       "B",
+							"metadata": map[string]interface{}{
+								"namespace": "ns",
+								"name":      "bar",
+								"uid":       "2",
+							},
+						},
+					},
+				},
+			},
+			want: wantGraph{
+				nodes: map[string]wantGraphItem{
+					"1": { // the object: not virtual (observed), without dependants
+						virtual:    false,
+						dependants: nil,
+					},
+					"2": { // the object owner: not virtual (observed), with 1 dependants
+						virtual:    false,
+						dependants: []string{"1"},
+					},
+				},
+			},
+		},
+		{
+			name: "Add an object with an owner ref and its owner (reverse discovery order)",
+			args: args{
+				objs: []*unstructured.Unstructured{
+					{
+						Object: map[string]interface{}{
+							"apiVersion": "b/v1",
+							"kind":       "B",
+							"metadata": map[string]interface{}{
+								"namespace": "ns",
+								"name":      "bar",
+								"uid":       "2",
+							},
+						},
+					},
+					{
+						Object: map[string]interface{}{
+							"apiVersion": "a/v1",
+							"kind":       "A",
+							"metadata": map[string]interface{}{
+								"namespace": "ns",
+								"name":      "foo",
+								"uid":       "1",
+								"ownerReferences": []interface{}{
+									map[string]interface{}{
+										"apiVersion": "b/v1",
+										"kind":       "B",
+										"name":       "bar",
+										"uid":        "2",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			want: wantGraph{
+				nodes: map[string]wantGraphItem{
+					"1": { // the object: not virtual, without dependants
+						virtual:    false,
+						dependants: nil,
+					},
+					"2": { // the object owner: not virtual (observed), with 1 dependants
+						virtual:    false,
+						dependants: []string{"1"},
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			gb := newObjectGraph(nil)
+			for _, o := range tt.args.objs {
+				gb.addObj(o)
+			}
+
+			assetGraph(t, gb, tt.want)
+		})
+	}
+}
+
+type objectGraphTestArgs struct {
+	objs []runtime.Object
+}
+
+var objectGraphsTests = []struct {
+	name    string
+	args    objectGraphTestArgs
+	want    wantGraph
+	wantErr bool
+}{
+	{
+		name: "Cluster",
+		args: objectGraphTestArgs{
+			objs: test.NewFakeCluster("ns1", "cluster1").Objs(),
+		},
+		want: wantGraph{
+			nodes: map[string]wantGraphItem{
+				"cluster.x-k8s.io/v1alpha3, Kind=Cluster, ns1/cluster1": {
```

Comment:
Shouldn't these be UIDs?

Reply:
They are actually UIDs, but in the test framework (rif. #2059) I made sure to create "human-readable, deterministic" strings instead of random strings

Reply:
That explains why I was so confused! I hadn't looked at 2059 yet.